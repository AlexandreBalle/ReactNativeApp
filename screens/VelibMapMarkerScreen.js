import React, { useState, useEffect, useContext } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import MapView from 'react-native-maps';
import { Icon } from 'react-native-elements';
import PropTypes from'prop-types';

const VelibMapMarkerScreen = ({ velib }) => {
  const coords = {
    latitude: velib.geometry.coordinates[1],
    longitude: velib.geometry.coordinates[0]
  };

  return (
    <MapView.Marker coordinate={coords}>
      <MapView.Callout tooltip style={styles.tooltip_container}>
        <View style={{backgroundColor: '#282c34'}}>
          <Text style={styles.title}>{velib.fields.station_name}</Text>
          <Text style={styles.description}>
          {
            "Nb de vélo dispo : " + velib.fields.nbfreedock + "/" + velib.fields.nbdock + "\n" +
            "Nb de vélo éléc. dispo : " + velib.fields.nbfreeedock + "/" + velib.fields.nbedock
          }
          </Text>
        </View>
      </MapView.Callout>
    </MapView.Marker>
  );
}

const styles = StyleSheet.create({
  tooltip_container: {
    width: 300,
    height: 100,
  },
  title: {
    color: "#61dafb",
    fontSize: 14,
    paddingLeft: 5,
  },
  description: {
    color: "white",
    fontSize: 12,
    paddingLeft: 5,
  },
});

VelibMapMarkerScreen.propTypes = {
  velib: PropTypes.shape({
    datasetid: PropTypes.string,
    recordid: PropTypes.string,
    fields: PropTypes.shape({
      station_state: PropTypes.string,
      maxbikeoverflow: PropTypes.number,
      densitylevel: PropTypes.string,
      nbbikeoverflow: PropTypes.number,
      dist: PropTypes.string,
      nbedock: PropTypes.number.isRequired,
      station_name: PropTypes.string.isRequired,
      kioskstate: PropTypes.string,
      nbfreeedock: PropTypes.number.isRequired,
      station_type: PropTypes.string,
      station_code: PropTypes.string,
      creditcard: PropTypes.string,
      station: PropTypes.string,
      nbfreedock:PropTypes.number.isRequired,
      duedate: PropTypes.string,
      nbebikeoverflow: PropTypes.number,
      nbebike: PropTypes.number,
      overflow: PropTypes.string,
      nbdock: PropTypes.number.isRequired,
      geo: PropTypes.array,
      overflowactivation: PropTypes.string,
      nbbike: PropTypes.number,
    }),
    geometry: PropTypes.shape({
      type: PropTypes.string,
      coordinates: PropTypes.array.isRequired
    }),
    record_timestamp: PropTypes.string
  })
};

export default VelibMapMarkerScreen;
