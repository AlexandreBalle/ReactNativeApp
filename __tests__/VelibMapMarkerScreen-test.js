import React from 'react';
import VelibMapMarkerScreen from '../screens/VelibMapMarkerScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const valueVelib = {
    "fields": {
      "nbedock": 21,
      "station_name": "Diderot - Pierre Bourdan",
      "nbfreeedock": 10,
      "nbfreedock": 0,
      "nbdock": 0,
    },
    "geometry": {
      "coordinates": [
        2.3902582749724393,
        48.84764862079675
      ]
    }
  };
  const tree = renderer.create(<VelibMapMarkerScreen velib={valueVelib} />).toJSON();
  expect(tree).toMatchSnapshot();
});
