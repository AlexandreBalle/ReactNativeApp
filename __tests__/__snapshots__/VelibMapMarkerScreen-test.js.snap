// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 48.84764862079675,
      "longitude": 2.3902582749724393,
    }
  }
  onPress={[Function]}
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
>
  <AIRMapCallout
    alphaHitTest={false}
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "backgroundColor": "white",
          "padding": 5,
        },
      ]
    }
    tooltip={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "padding": 5,
        }
      }
    >
      <Text>
        Diderot - Pierre Bourdan
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "paddingTop": 5,
          }
        }
      >
        Nb de vélo dispo : 0/0
Nb de vélo éléc. dispo : 10/21
      </Text>
    </View>
  </AIRMapCallout>
</AIRMapMarker>
`;
